### ISSUE TITLE
Understanding Training Parameters for 2xTesla P40

### ISSUE THREAD

**Issue by dexception** [AlexeyAB/darknet#4242]
Training with following hardware configuration:
1. 2xTesla P40
2. Samsung Evo 860
3. Intel(R) Xeon(R) Silver 4110 CPU @ 2.10GHz
4. 64 GB RAM

batch=64
subdivisions=2
=========================================
I am training on 8 classes.
=========================================
 (next mAP calculation at 176148 iterations) 
 171163: 2.237339, 2.325732 avg loss, 0.001471 rate, 2.584813 seconds, 21908864 images

Syncing seems to be taking more time when i increase the batch(64/2) value. 

Also i don't seem to understand how many iterations i need to train for getting decent accuracy as the number is vastly different for yolov3-pan model. For a single class i trained it for about 800,000 iterations for getting the max accuracy. So for 8 classes do i have to train till 8x800,000 times. That seem alot. Can you please explain how much iterations one needs to train because it can also depend upon batch size and model type. 

Thanks.

**Comment by AlexeyAB** [comment]
> So for 8 classes do i have to train till 8x800,000 times. 

Nobody knows this.
If you use batch=64, then usually you should train as many iterations as images in your training dataset, or more.
Just train while the mAP increases: https://github.com/AlexeyAB/darknet#when-should-i-stop-training
and get the best weights-file.

Do you use this cfg-file https://github.com/AlexeyAB/darknet/files/3253820/yolo_v3_spp_pan_scale.cfg.txt
with `policy=steps` ?

**Comment by dexception** [comment]
I am using the one with policy=sgdr.
Can you elaborate the difference between these two ?

**Comment by AlexeyAB** [comment]
Try to use 
```
policy=steps
steps=8000,9000
scales=.1,.1
```


----


About SGDR: https://towardsdatascience.com/https-medium-com-reina-wang-tw-stochastic-gradient-descent-with-restarts-5f511975163

and: https://arxiv.org/abs/1608.03983v5


About parameters `sgdr_cycle, sgdr_mult, ...`  : https://github.com/AlexeyAB/darknet/wiki/CFG-Parameters-in-the-%5Bnet%5D-section
